libtenzir/builtins/formats/xsv.cpp

```diff
@@ -245,22 +248,25 @@ auto parse_impl(generator<std::optional<std::string_view>> lines,
   // Parse header.
   auto it = lines.begin();
   auto header = std::optional<std::string_view>{};
-  while (it != lines.end()) {
-    auto line = *it;
-    ++it;
-    if (not line) {
-      co_yield {};
-      continue;
+  if (args.header) {
+    header = *args.header;
+  } else
+    while (it != lines.end()) {
+      auto line = *it;
+      ++it;
+      if (not line) {
+        co_yield {};
+        continue;
+      }
+      if (line->empty())
+        continue;
+      if (args.allow_comments && line->front() == '#')
```

Comment:
Just seeing that the comment char is still hard-coded, but that's a separate thing.

Reply:
Let's wait until someone comes with a request. Not for this PR.